e2e/test_operator_upgrades/operator_upgrades_test.go

```diff
@@ -650,4 +650,75 @@ var _ = Describe("Operator Upgrades", Label("e2e", "pr"), func() {
 		EntryDescription("Upgrade from %[1]s to %[2]s"),
 		fixtures.GenerateUpgradeTableEntries(testOptions),
 	)
+
+	DescribeTable(
+		"one process cannot connect to the Kubernetes API",
+		func(beforeVersion string, targetVersion string) {
+			if !factory.ChaosTestsEnabled() {
+				Skip("Chaos tests are skipped for the operator")
+			}
+
+			if fixtures.VersionsAreProtocolCompatible(beforeVersion, targetVersion) {
+				Skip("this test only affects version incompatible upgrades")
+			}
+
+			// If we are not using the unified image, we can skip this test.
+			if !factory.UseUnifiedImage() {
+				Skip("The sidecar image doesn't require connectivity to the Kubernetes API")
+			}
+
+			clusterSetup(beforeVersion, true)
+
+			selectedPod := factory.RandomPickOnePod(fdbCluster.GetStoragePods().Items)
+
+			var kubernetesServiceHost string
+			Eventually(func(g Gomega) error {
+				std, _, err := factory.ExecuteCmdOnPod(
+					&selectedPod,
+					fdbv1beta2.MainContainerName,
+					"printenv KUBERNETES_SERVICE_HOST",
+					false,
+				)
+
+				g.Expect(std).NotTo(BeEmpty())
+				kubernetesServiceHost = strings.TrimSpace(std)
+
+				return err
+			}, 5*time.Minute).ShouldNot(HaveOccurred())
+
+			exp := factory.InjectPartitionWithExternalTargets(fixtures.PodSelector(&selectedPod), []string{kubernetesServiceHost})
+			// Make sure that the partition takes effect.
+			Eventually(func() error {
+				_, _, err := factory.ExecuteCmdOnPod(
+					&selectedPod,
+					fdbv1beta2.MainContainerName,
+					fmt.Sprintf("nc -vz -w 2 %s 443", kubernetesServiceHost),
+					false,
+				)
+
+				return err
+			}).WithTimeout(2 * time.Minute).WithPolling(1 * time.Second).Should(HaveOccurred())
+
+			// Update the cluster version.
+			Expect(fdbCluster.UpgradeCluster(targetVersion, false)).NotTo(HaveOccurred())
+
+			if !fixtures.VersionsAreProtocolCompatible(beforeVersion, targetVersion) {
```

Comment:
just for my understanding, if they are protocol compatible then what changes with the upgrade? wouldn't we still expect the same behaviour (except that the IncorrectConfig conditions are not added, and so the bounce isn't needed)? From my reading, it seems like the only thing that would change is the wait for config maps to update before the coordinated bounce is done to upgrade the cluster, but I seem to be missing something.

Reply:
When the cluster is upgraded to a protocol compatible version (patch upgrade) the operator skips the staging and restart phase and directly goes to the [recreation](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/main/docs/manual/upgrades.md#recreation-of-pods-phase) phase as all fdbserver processes will be able to communicate, so there is no need to stage the new binaries and that's the only step that blocks. But I agree that we should test this behaviour too.

Reply:
ohh so since there is no blocking / waiting for configmap update, there isn't need for the k8s API just FDB cluster communication?

Reply:
Correct 👍 And once the Pod with the K8s API connection issues is recreated the chaos injection will be lost.

Reply:
okay that makes sense, thanks!